tox.ini

```diff
@@ -5,6 +5,24 @@ exclude=*certdata*,*manifestdata*
 jobs=auto
 max-line-length=300
 
-[nosetests]
-with-id=True
-attr='!functional'
+[pytest]
+addopts =
+	# Non-registered markers raise errors
+	--strict-markers
+	# Start with tests that failed last time, --ff
+	--failed-first
+	# Show extra summary for (f)ailed, (E)rror, (s)kipped and (w)arnings
+	-rfEsw
+	# Mark some tests not to be run
+	-m "not zypper and not dbus and not functional"
+markers =
+	dbus: subscription-manager tests for DBus.
+	slow: subscription-manager tests that may be slower than the rest.
+	zypper: subscription-manager tests for the Zypper package manager.
+	functional: subscription-manager functional tests.
+timeout = 5
+testpaths =
+	test/
+required_plugins =
+	pytest-randomly
+	pytest-timeout
```

Comment:
To be honest, we IMHO don't use `tox` at all. Could it be useful for anything?

Reply:
We do not, but most of the tools try to read their configuration from here. IMHO it's better to keep them all in one place instead of having them all separated in their own configs (which we are currently doing for flake8 with its `.flake8`).

We can use `.pytest.ini` as a source of configuration for pytest, but that's another file in the project root. In this sense I'd vote to move the flake8 config here to the `tox.ini` in near future, to keep them all together.

Reply:
personally, i'd prefer to have each tool using its own configuration file, as it makes easier to see where the configuration for each lives; i already made the mistake once with `.flake8`...